### PR TITLE
Add auto battle resolve mode to the game

### PR DESCRIPTION
**Pull request by idshibanov**
Sped up AI battles as well by removing unnecessary delay, now it happens almost instantly.

Added auto-combat and auto spellcasting settings and included them in base interface:
image

image

I hope this will be useful for many, especially people debugging the game.

### CODE REVIEW

File: src/fheroes2/system/settings.cpp [hunk 1]
```diff
@@ -1051,6 +1051,26 @@ void Settings::SetBattleSpeed( int speed )
     battle_speed = speed;
 }
 
+void Settings::setBattleAutoResolve( bool enable )
```
Comment (ihhub) [inline src/fheroes2/system/settings.cpp:1054]: These settings should be a part of `opt_global` as they are switchable inside UI settings. Also please add them for reading and writing into the condiguration file.

File: src/fheroes2/system/settings.cpp [hunk 2]
```diff
@@ -1485,6 +1505,16 @@ void Settings::ExtResetModes( u32 f )
     }
 }
 
+bool Settings::ExtBattleAutoResolve( void ) const
```
Comment (ihhub) [inline src/fheroes2/system/settings.cpp:1508]: The name of the methods should be different if we move them into global settings.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @idshibanov , could you please move new settings to global as this is the best place for them?

File: src/fheroes2/system/settings.cpp (hunk 1 above)
Comment (idshibanov) [reply src/fheroes2/system/settings.cpp:1071]: Updated.

File: src/fheroes2/battle/battle_main.cpp [hunk 3]
```diff
@@ -83,17 +83,18 @@ Battle::Result Battle::Loader( Army & army1, Army & army2, s32 mapsindex )
             army2.GetCommander()->ActionPreBattle();
     }
 
-    bool local = army1.isControlHuman() || army2.isControlHuman();
+    const bool isHumanBattle = army1.isControlHuman() || army2.isControlHuman();
+    bool showBattle = !Settings::Get().BattleAutoResolve() && isHumanBattle;
```
Comment (ihhub) [inline src/fheroes2/battle/battle_main.cpp:87]: We shouldn't allow instant battle for both armies being human-controlled. For example, in multiplayer.

Comment (ihhub) [reply src/fheroes2/battle/battle_main.cpp:87]: So the logic should be like this:
```
const bool autoresolveBattle = Settings::Get().BattleAutoResolve();
const bool isHumanBattle = army1.isControlHuman() || army2.isControlHuman();
const bool allHumans = army1.isControlHuman() && army2.isControlHuman();
const bool showBattle = allHumans || ( !autoresolveBattle && isHumanBattle );
```

**Review by ihhub — COMMENTED** [review]
Hi @idshibanov , I commented only one place. Could you please take a look?

**Comment by ihhub** [comment]
@idshibanov , great work! Thank you very much for such feature!

**Pull request merged**